hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerStateMachine.java

```diff
@@ -162,6 +167,10 @@ public void notifyLeaderChanged(RaftGroupMemberId groupMemberId,
     // Initialize OMHAMetrics
     ozoneManager.omHAMetricsInit(newLeaderId.toString());
     LOG.info("{}: leader changed to {}", groupMemberId, newLeaderId);
+    // if the node is leader (can be ready or not ready, need update index)
+    if (ozoneManager.getOmRatisServer().checkLeaderStatus() != OzoneManagerRatisServer.RaftServerStatus.NOT_LEADER) {
+      indexGenerator.onLeaderChange();
```

Comment:
As every OM has its IndexGenerator, which works independently, could you explain a little bit here, what the onLeaderChange actually do?

Reply:
Ratis generate termIndex for below cases:
1. transaction update via applyTransaction()
2. notifyConfigChange for case of ratis internal book-keeping

IndexGenerate will generate index in pre-ratis execution for every incoming request at leader node.

So both can have different value, where mostly ratis index will be faster side.

OM will use OM index for purpose of objectId and updateId generation as used currently.
TermIndex will be used to handle replay of ratis log pending, and ensure nodes are in sync in upgrade prepare cases.

So use case will be separated and hence will not have impact.

Reply:
@ChenSammi I got the problem,
Since every node generate the index on request execution, so there is a possibility of different index value. All nodes should get index from common place in this case, will work over this to handle.